MetaMask/metamask-extension
# fix(mme-14830): fix consistent tab when switching languages

## Explanation

**Fixes #14830**

The `<Tab />` component is currently taking a translated `string` as unique key to render the DOM. Hence when we switch language and land in home page again, it is trying to look for the responding `string` from previous language from the new language strings set. 
The fix will be to have `Tab` component to take the `key` in `message.json` file and map by key. Each time a new action `setDefaultHomeActiveTabName` is dispatched, it will pass the `key` (eg: `asset`, `activity`) as props to child component.

*Moreover, for the convenience of reviewing, what I have done in this fix are:* 
-   Fix the bug, which comes from Tabs & Tab component, to use key(which is the same as language file message key) instead of name as index
-   Change the props name for Tabs and related function signature to make it more clear that we are using key instead of string (name)
-   Adapt related storybook

*What I didn’t do*
-   No test added, as there’s no test for home page :(
-   Didn’t change the state and action name, as it will reach too many files and make MR too complex to review, I would like to focus on bug fix this time


## Screenshots/Screencaps



### Before

https://user-images.githubusercontent.com/12678455/212040399-6a7c2587-9490-4215-8713-298c74aa89b2.mp4


### After


https://user-images.githubusercontent.com/12678455/212382192-392aa6c3-ad14-4742-b6fd-244209eaed31.mov



## Manual Testing Steps

1.  Unlock Metamask
2.  Go to Settings
3.  Change language to another different than English
4.  Perform a transaction
5.  Notice that you land on the Home/Assets page instead of Home/Activity page

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
